src/components/dropdown/UIDropdown.vue

```diff
@@ -151,34 +153,51 @@ export default defineComponent({
                 const maxHeight = getPxValue(props.width);
 
                 // doesn't fit on the right
-                if (globalThis?.document.documentElement.clientWidth - (contentParameters.x + mousePos.x) <= width) {
-                    style.left = `${mousePos.x - width}px`;
+                if (globalThis?.document.documentElement.clientWidth - (dropdownRect.x + mousePos.x) <= width) {
+                    style.left = `${dropdownRect.left + mousePos.x - width}px`;
                 } else {
-                    style.left = `${mousePos.x}px`;
+                    style.left = `${dropdownRect.left + mousePos.x}px`;
                 }
 
+                const top = dropdownRect.top + mousePos.y;
                 // doesn't fit on the bottom
-                if (
-                    globalThis?.document.documentElement.clientHeight - (contentParameters.y + mousePos.y) <= maxHeight
-                ) {
-                    style.top = `${mousePos.y - (dropdownParameters ? dropdownParameters.height : maxHeight)}px`;
+                if (globalThis?.document.documentElement.clientHeight - top <= maxHeight) {
+                    style.top = `${top - (contentRect ? contentRect.height : maxHeight)}px`;
                 } else {
-                    style.top = `${mousePos.y}px`;
+                    style.top = `${top}px`;
                 }
 
                 return style;
             }
 
+            const contentHeight = contentRect ? contentRect.height : 0;
+            const contentWidth = contentRect ? contentRect.width : 0;
+
             if (props.horizontal === 'left') {
-                style.left = '0px';
+                style.left = `${dropdownRect.left}px`;
             } else {
-                style.right = '0px';
+                style.left = `${dropdownRect.left + dropdownRect.width - contentWidth}px`;
             }
 
+            // todo - if there's no space to show the content in the current clientHeight(it can still scroll)
+            //  anchor to bottom of the screen
             if (props.vertical === 'top') {
-                style.bottom = `${contentParameters.height}px`;
+                const position = globalThis.document.documentElement.scrollTop
+                    + dropdownRect.top
+                    - dropdownRect.height / 2
+                    - contentHeight;
+                // if negative value, anchor to top of window
+                style.top = `${Math.max(0, position)}px`;
             } else {
-                style.top = `${contentParameters.height}px`;
+                let top = globalThis?.document.documentElement.scrollTop
+                    + dropdownRect.top
+                    + dropdownRect.height;
+
+                if (globalThis?.document.documentElement.scrollHeight <= top + contentHeight) {
+                    top = globalThis?.document.documentElement.scrollHeight - contentHeight;
+                }
+
+                style.top = `${top}px`;
```

Comment:
Regression here: menu no longer aligns itself with the parent when using `vertical="top"` (image shows right-click menu).
 
![image](https://user-images.githubusercontent.com/14146557/192735088-29af75f8-7588-4c8b-a3f1-a717f9716e0d.png)